### PR TITLE
Bugfix – Inferred cell type field not in SDRF breaks experiment design

### PR DESCRIPTION
**Pull request by alfonsomunozpomer**
If inferred cell type fields are not present in the SDRF file (e.g. E-CURD-4) then the script assigns a null value to the experiment design column ordinal.

In such cases, the inferred cell type column should be placed in the last position (highest ordinal plus one).

Things I did here:
- Utility/Common functions were spread between two scripts, I consolidated into just `common_routines.sh`
- I extracted the AWK code into a separate file that’s easier to understand and work with (maybe even test!)
- Rather than piping each SQL `INSERT` to `psql` all the statements are written to a file, enclosed in a transaction block **and then** have `psql` process it
- Moved all the fixtures to the `tests` directory (remember: the `fixtures` directory is for the scripts that we use to generate fixtures for `atlas-web-single-cell`)
- Replaced the condensed SDRF to test the experiment design load to use a file where a field isn’t present in the SDRF file
- Lastly, I updated the image used in the tests, which is a story for another time

### CODE REVIEW

**Review by ke4 — APPROVED** [review]
Looks fine with me 👍

File: bin/delete_collection.sh [hunk 1]
```diff
@@ -3,7 +3,6 @@
 set -e
 
 scriptDir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-source $scriptDir/db_scxa_common.sh
```
Comment (anilthanki) [inline bin/delete_collection.sh:6]: Just checking don't you need `source common_routines.sh` here?

File: bin/get_experiment_info.sh [hunk 2]
```diff
@@ -3,7 +3,6 @@
 set -e
 
 scriptDir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-source $scriptDir/db_scxa_common.sh
```
Comment (anilthanki) [inline bin/get_experiment_info.sh:6]: Just checking don't you need `source common_routines.sh` here?

File: bin/delete_collection.sh (hunk 1 above)
Comment (alfonsomunozpomer) [reply bin/delete_collection.sh:6]: It’s explained in the PR description above. However, this script isn’t using any imported functions.

File: bin/get_experiment_info.sh (hunk 2 above)
Comment (alfonsomunozpomer) [reply bin/get_experiment_info.sh:6]: It’s explained in the PR description above. However, this script isn’t using any imported functions.

File: run_tests_with_containers.sh [hunk 3]
```diff
@@ -1,8 +1,5 @@
 #!/usr/bin/env bash
 
-export SOLR_HOST=my_solr:8983
-export ZK_HOST=gxa-zk-1
-export ZK_PORT=2181
```
Comment (anilthanki) [inline run_tests_with_containers.sh:5]: Aren't they necessary anymore?

Comment (anilthanki) [reply run_tests_with_containers.sh:5]: Ignore this commentt

**Review by anilthanki — APPROVED** [review]
LGTM

**Comment by pmb59** [comment]
Hi @alfonsomunozpomer can you please rebase to include this https://github.com/ebi-gene-expression-group/db-scxa/pull/68

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM quay.io/ebigxa/atlas-db-scxa-base:0.1.0
+FROM quay.io/ebigxa/atlas-db-scxa-base:0.15.0.0
```
Comment (pmb59) [inline Dockerfile:1]: This tag doesn't have the best naming - but it does exist in quay so let's leave it as it is

File: bin/add_exps_to_collection.sh [hunk 5]
```diff
@@ -3,7 +3,6 @@
 set -e
 
 scriptDir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-source $scriptDir/db_scxa_common.sh
```
Comment (pmb59) [inline bin/add_exps_to_collection.sh:6]: Should't you have here replacing the line you've removed? 

```suggestion
source $scriptDir/common_routines.sh
```

**Review by pmb59 — APPROVED** [review]
LGTM @alfonsomunozpomer !

I have a small request related to 

> Replaced the condensed SDRF to test the experiment design load to use a file where a field isn’t present in the SDRF file

New experiment for test is E-CURD-4 (Arabidopsis). Could we still keep the previous one E-MTAB-2983 (human) as a separate test in the .bats file?

File: bin/add_exps_to_collection.sh (hunk 5 above)
Comment (ke4) [reply bin/add_exps_to_collection.sh:6]: no, as we are not using any functions from that shared file

File: Dockerfile (hunk 4 above)
Comment (alfonsomunozpomer) [reply Dockerfile:1]: I decided to keep it in sync with major releases of Single Cell Expression Atlas (so that v15 corresponds to `0.15.0.0`).

**Comment by alfonsomunozpomer** [comment]
> LGTM @alfonsomunozpomer !
> 
> I have a small request related to
> 
> > Replaced the condensed SDRF to test the experiment design load to use a file where a field isn’t present in the SDRF file
> 
> New experiment for test is E-CURD-4 (Arabidopsis). Could we still keep the previous one E-MTAB-2983 (human) as a separate test in the .bats file?

Done.

**Pull request merged**